### PR TITLE
add 0-robot template

### PR DESCRIPTION
**Pull request by alichaddad**

### CODE REVIEW

File: templates/zrobot/zrobot.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import ast
+from js9 import j
+from zerorobot.template.base import TemplateBase
+from zerorobot.template.state import StateCheckError
+
+
+#Temporary flist needs to be updated to offical 0-robot flist
+FLIST_ZROBOT = 'https://hub.gig.tech/fastgeert/jumpscale-0-robot-latest.flist'
```
Comment (grimpy) [inline templates/zrobot/zrobot.py]: we should have it in official apps

File: templates/zrobot/zrobot.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+
+class Zrobot(TemplateBase):
+
+    version = '0.0.1'
+    template_name = "zrobot"
+
+    def __init__(self, name, guid=None, data=None):
+        super().__init__(name=name, guid=guid, data=data)
+        self._validate_input()
```
Comment (grimpy) [inline templates/zrobot/zrobot.py]: use buildint validate method instead

File: templates/zrobot/zrobot.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        zrobot_sal = self._get_zrobot_client(contservice)
+        zrobot_sal.start(port=6600, template_repos=self.data['templates'])
+        self.state.set('actions', 'install', 'ok')
+
+    def uninstall(self):
+        self.state.check('actions', 'install', 'ok')
+        contservice = self._get_container()
+        zrobot_sal = self._get_zrobot_client(contservice)
+        zrobot_sal.stop()
+        self.state.delete('actions', 'install')
```
Comment (grimpy) [inline templates/zrobot/zrobot.py]: this should probably nuke the container service created aswell

File: templates/zrobot/zrobot.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+class Zrobot(TemplateBase):
+
+    version = '0.0.1'
+    template_name = "zrobot"
+
+    def __init__(self, name, guid=None, data=None):
+        super().__init__(name=name, guid=guid, data=data)
+        self.validate()
+    
+    def validate(self):
```
Comment (serboctor) [inline templates/zrobot/zrobot.py]: change `validate` to `_validate_input`

File: templates/zrobot/zrobot.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        if not nics:
+            nics = [{'type': 'default'}]
+            ports = ['6600:6600']
+
+        data = {
+            'node': self.data['node'],
+            'flist': FLIST_ZROBOT,
+            'nics': nics,
+            'hostname': self.name,
+            'hostNetworking': False,
```
Comment (serboctor) [inline templates/zrobot/zrobot.py]: it will be false by default, you don't need to set it

**Comment by serboctor** [comment]
Fixes https://github.com/zero-os/0-templates/issues/53

**Pull request merged**